glotaran/io/preprocessor/preprocessor.py

```diff
@@ -0,0 +1,78 @@
+"""A pre-processor pipeline for data."""
+from __future__ import annotations
+
+import abc
+from typing import Literal
+
+import xarray as xr
+from pydantic import BaseModel
+
+
+class PreProcessor(BaseModel, abc.ABC):
+    """A base class for pre=processors."""
+
+    class Config:
+        """Config for BaseModel."""
+
+        arbitrary_types_allowed = True
+
+    @abc.abstractmethod
+    def apply(self, data: xr.DataArray) -> xr.DataArray:
+        """Apply the pre-processor.
+
+        Parameters
+        ----------
+        data: xr.DataArray
+            The data to process.
+
+        Returns
+        -------
+        xr.DataArray
+
+        .. # noqa: DAR202
+        """
+
+
+class CorrectBaselineValue(PreProcessor):
+    """Corrects a dataset by subtracting baseline value."""
+
+    action: Literal["baseline-value"] = "baseline-value"
+    value: float
+
+    def apply(self, data: xr.DataArray) -> xr.DataArray:
+        """Apply the pre-processor.
+
+        Parameters
+        ----------
+        data: xr.DataArray
+            The data to process.
+
+        Returns
+        -------
+        xr.DataArray
+        """
+        return data - self.value
+
+
+class CorrectBaselineAverage(PreProcessor):
+    """Corrects a dataset by subtracting the average over a part of the data."""
+
+    action: Literal["baseline-average"] = "baseline-average"
+    interval: dict[str, slice]
+
+    def apply(self, data: xr.DataArray) -> xr.DataArray:
+        """Apply the pre-processor.
+
+        Parameters
+        ----------
+        data: xr.DataArray
+            The data to process.
+
+        Returns
+        -------
+        xr.DataArray
+        """
+        selection = data
+        for axis, interval in self.interval.items():
+            selection = selection.sel({axis: interval})
+        return data - (selection.sum() / selection.size)
```

Comment:
This code needs some revision, it suggests that it works with more than one intervals, but it doesn't.

Suppose an time axis [0,100], are you intending it to work, [0,10] and [90,100] in one operations, or only [0,10] and then [90,100] in two operations. It's not clear, and there's no tests for this either.

Reply:
I also added `exclude`, if you eg want to average only the edges.

Reply:
The code is more correct and clear now than before.
```python
return data - data.sel(self.select or {}).drop_sel(self.exclude or {}).mean()
```